specs/agents/universal-profiling-integration.md

```diff
@@ -83,6 +83,8 @@ transaction-id        | uint8[8]
 * *span-id*: The W3C trace id of the currently active span
 * *transaction-id*: The W3C span id of the currently active transaction (=the local root span)
 
+APM-agents MAY start populating the thread-local storage only after receiving a host agent [registration message](#profiler-registration-message)
```

Comment:
The use case would be if you collect traces for the application startup and what to have profiling data for those at the very beginning:
There will be a short delay between application startup and receiving this initial registration message. As a result, spans started and activated before that receival would not get correlated.
This can be overcome for those edge cases by eagerly populating the TLS, even if it is not known whether a profiler is already there.

I'm planning to implement this by having a tri-state `enabled` config option:
 * `false`: No correlation, the native lib won't even be loaded
 * `true`: Correlation active, TLS will be eagerly populated
 * `auto` (default for OTel-extension): Correlation will be active, TLS will only be populated after receiving the profiler registration message (So basically close to zero overhead if no profiler is active)